SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp

```diff
@@ -492,22 +493,33 @@ AActor* USpatialReceiver::CreateActor(improbable::Position* Position, improbable
 
 void USpatialReceiver::ApplyComponentData(Worker_EntityId EntityId, Worker_ComponentData& Data, USpatialActorChannel* Channel)
 {
-	UClass* Class = TypebindingManager->FindClassByComponentId(Data.component_id);
-	checkf(Class, TEXT("Component %d isn't hand-written and not present in ComponentToClassMap."), Data.component_id);
+	uint32 Offset = 0;
+	bool bFoundOffset = TypebindingManager->FindOffsetByComponentId(Data.component_id, Offset);
+	if (!bFoundOffset)
+	{
+		return;
+	}
 
-	UObject* TargetObject = GetTargetObjectFromChannelAndClass(Channel, Class);
-	if (!TargetObject)
+	UObject* TargetObject = PackageMap->GetObjectFromUnrealObjectRef(FUnrealObjectRef(EntityId, Offset));
```

Comment:
Any reason to get it from the package map instead of the subobject property?

Reply:
You need to get the TargetObject. If the TargetObject is an actor, then you use the package map to get it. If the TargetObject is a subobject, and we want to use properties to get the right object, you still have to use the package map to get the Actor and then use the property, when we could have just used the package map to get the subobject. I am going to leave it as is.